src/breakupModel/simulation/Breakup.cpp

```diff
@@ -1,110 +1,101 @@
-
     std::uniform_real_distribution<> uniformRealDistribution{0.0, 1.0};
-    std::for_each(_output.begin(),
-                  _output.end(),
-                  [&](Satellite &sat) {
-                      const double y = uniformRealDistribution(_randomNumberGenerator);
-                      const double L_c = transformUniformToPowerLaw(_minimalCharacteristicLength,
-                                                                    _maximalCharacteristicLength,
-                                                                    powerLawExponent,
-                                                                    y);
-                      sat.setCharacteristicLength(L_c);
-                  });
+    std::for_each(std::execution::par, _output._characteristicLength.begin(), _output._characteristicLength.end(),
+                  [&](double &lc) {
+        const double y = getRandomNumber(uniformRealDistribution);
+        lc = transformUniformToPowerLaw(_minimalCharacteristicLength, _maximalCharacteristicLength, powerLawExponent, y);
+    });
 }
 
 void Breakup::areaToMassRatioDistribution() {
-    auto satIt = _output.begin();
-    for (; satIt != _output.end(); ++satIt) {
-        const double lc = satIt->getCharacteristicLength();
-
+    auto tupleView = _output.getAreaMassTuple();
+    std::for_each(std::execution::par, tupleView.begin(), tupleView.end(),
+                  [&](auto &tuple) {
+        //Order in the tuple: 0: L_c | 1: A/M | 2: Area | 3: Mass
         //Calculate the A/M value in [m^2/kg]
-        const double areaToMassRatio = calculateAM(lc);
+        std::get<1>(tuple) = calculateAM(std::get<0>(tuple));
 
         //Calculate the are A in [m^2]
-        double area = 0;
-        if (lc < 0.00167) {
-            area = 0.540424 * lc * lc;
+        if (std::get<0>(tuple) < 0.00167) {
+            std::get<2>(tuple) = 0.540424 * std::get<0>(tuple) * std::get<0>(tuple);
         } else {
-            area = 0.556945 * std::pow(lc, 2.0047077);
+            std::get<2>(tuple) = 0.556945 * std::pow(std::get<0>(tuple), 2.0047077);
         }
-
         //Calculate the mass m in [kg]
-        const double mass = area / areaToMassRatio;
-
-        //Finally set every value in the satellite
-        satIt->setAreaToMassRatio(areaToMassRatio);
-        satIt->setArea(area);
-        satIt->setMass(mass);
-
-        //Stop calculating further values if the produced mass already exceeds the input mass
-        //and erases the elements which are no longer needed
-        _outputMass += mass;
-        if (_outputMass > _inputMass) {
-//            spdlog::warn("The simulation reduced the number of fragments because the mass budget was exceeded. "
-//                         "In other words: The random behaviour has produced heavier fragments");
-            _outputMass -= mass;
-            _output.erase(satIt, _output.end());
-            break;
-        }
+        std::get<3>(tuple) = std::get<2>(tuple) / std::get<1>(tuple);
```

Comment:
the previous code was much more readable. maybe this can help?

```suggestion
        // mass = area / areaToMassRatio
        std::get<3>(tuple) = std::get<2>(tuple) / std::get<1>(tuple);
```